test/Microsoft.AspNet.Mvc.Razor.Test/RazorViewEngineTest.cs

```diff
@@ -507,6 +512,63 @@ public void FindView_LooksForViewsIfCachedViewDoesNotExist()
             expander.Verify();
         }
 
+        [Fact]
+        public void FindView_DoesNotInvokeExpandViewLocations_IfCacheEntryMatchesButViewIsNotFound()
```

Comment:
I'd add a note here that this validates an important perf concern. It's easy to revisit these tests in a while and thing "gee that's a complicted test... do we need to keep it?"